pkg/kubectl/cmd/util/factory.go

```diff
@@ -1235,3 +1273,72 @@ func (f *Factory) NewBuilder(thirdPartyDiscovery bool) *resource.Builder {
 
 	return resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true))
 }
+
+// TODO find a better place for dynamicObjectTyper
```

Comment:
I am hesitant to add it there as it would make meta depend on discovery client. I can move the type of the parameter(`discovery.APIGroupResources`) to the rest of the discovery types and then move this. Does that sound ok?